rclpy/src/rclpy_common/src/common.cpp

```diff
@@ -40,4 +42,51 @@ convert_to_py_names_and_types(const rcl_names_and_types_t * names_and_types)
   return py_names_and_types;
 }
 
+std::unique_ptr<void, destroy_ros_message_signature *>
+create_from_py(py::object pymessage)
+{
+  typedef void * create_ros_message_signature (void);
+
+  py::object pymetaclass = py::getattr(pymessage, "__class__");
```

Comment:
`pymessage.attr("__class__")` works too, but the free function style is fine if you like it.

Reply:
See https://github.com/ros2/rclpy/pull/721/commits/bc2a249968e3ad192e43f1679db2a74283d101fa.